### PR TITLE
Implement Log on top of logback without slf4j

### PR DESCRIPTION
**Pull request by dfahritdinov**

### CODE REVIEW

File: core/src/main/scala/com/evolutiongaming/catshelper/Log.scala [hunk 1]
```diff
@@ -134,6 +134,38 @@ object Log {
...
+    import ch.qos.logback.classic.Level
+    import ch.qos.logback.classic.spi.LoggingEvent
+
+    import scala.collection.JavaConverters._
+
+    val FQCN = getClass.getName
+
+    def append(msg: => String, mdc: Mdc, level: Level, throwable: Throwable = null): F[Unit] = Sync[F].delay {
+      val event = new LoggingEvent(FQCN, logger, level, msg, throwable, null)
+      mdc.context.map(_.toSortedMap) foreach { mdc =>
```
Comment (t3hnar) [inline core/src/main/scala/com/evolutiongaming/catshelper/Log.scala:148]: could it be too expensive to call `toSortedMap` on every log call?

Comment (t3hnar) [reply core/src/main/scala/com/evolutiongaming/catshelper/Log.scala:148]: what if logging is disabled ?

File: core/src/test/scala/com/evolutiongaming/catshelper/LogSpec.scala [hunk 2]
```diff
@@ -78,6 +78,16 @@ class LogSpec extends AnyFunSuite with Matchers {
 
     io.unsafeRunSync() shouldEqual null
   }
+
+  test("logback implementation") {
+    val io = for {
+      logOf <- LogOf.logback[IO]
+      log <- logOf(getClass)
+      _ <- log.info("hello from logback", Log.Mdc("k" -> "test value for K"))
```
Comment (t3hnar) [inline core/src/test/scala/com/evolutiongaming/catshelper/LogSpec.scala:86]: what are we testing here? :)

File: build.sbt [hunk 3]
```diff
@@ -51,8 +51,8 @@ lazy val core = project
       Cats.effect,
       machinist,
       `slf4j-api`,
+      logback,
```
Comment (t3hnar) [inline build.sbt:54]: so logbook is now required for all? would we have any problems with that?

**Review by t3hnar — CHANGES_REQUESTED** [review]

File: core/src/main/scala/com/evolutiongaming/catshelper/Log.scala (hunk 1 above)
Comment (dfahritdinov) [reply core/src/main/scala/com/evolutiongaming/catshelper/Log.scala:148]: Shouldn't be expensive as it's just cast: `asInstanceOf`

File: core/src/test/scala/com/evolutiongaming/catshelper/LogSpec.scala (hunk 2 above)
Comment (dfahritdinov) [reply core/src/test/scala/com/evolutiongaming/catshelper/LogSpec.scala:86]: possibility to log message and see it as output. Do you know better way to test such logger?

File: build.sbt (hunk 3 above)
Comment (dfahritdinov) [reply build.sbt:54]: I don't expect any issues as we're already using logback as implementation for slf4j

File: core/src/test/scala/com/evolutiongaming/catshelper/LogSpec.scala (hunk 2 above)
Comment (t3hnar) [reply core/src/test/scala/com/evolutiongaming/catshelper/LogSpec.scala:86]: register custom appenders from code in test with vars and check those are assigned to correct values?

**Pull request merged**